src/devices/Vl53L1X/Vl53L1X.cs

```diff
@@ -0,0 +1,1022 @@
+                            break;
+                        case TimingBudget.Budget100:
+                            WriteUInt16((ushort)Registers.RANGE_CONFIG__TIMEOUT_MACROP_A_HI, 0x01CC);
+                            WriteUInt16((ushort)Registers.RANGE_CONFIG__TIMEOUT_MACROP_B_HI, 0x01EA);
+                            break;
+                        case TimingBudget.Budget200:
+                            WriteUInt16((ushort)Registers.RANGE_CONFIG__TIMEOUT_MACROP_A_HI, 0x02D9);
+                            WriteUInt16((ushort)Registers.RANGE_CONFIG__TIMEOUT_MACROP_B_HI, 0x02F8);
+                            break;
+                        case TimingBudget.Budget500:
+                            WriteUInt16((ushort)Registers.RANGE_CONFIG__TIMEOUT_MACROP_A_HI, 0x048F);
+                            WriteUInt16((ushort)Registers.RANGE_CONFIG__TIMEOUT_MACROP_B_HI, 0x04A4);
+                            break;
+                        default:
+                            throw new ArgumentException(
+                                $"A timing budget of {TimingBudget.Budget15} can only be set in distance mode {DistanceMode.Short}.",
+                                nameof(budget));
+                    }
+
+                    break;
+            }
+        }
+
+        /// <summary>
+        ///     This function returns the current <see cref="TimingBudget" /> in ms.
+        /// </summary>
+        /// <returns>The current <see cref="TimingBudget" /></returns>
+        public TimingBudget GetTimingBudgetInMs()
+        {
+            ushort temp = ReadUInt16((ushort)Registers.RANGE_CONFIG__TIMEOUT_MACROP_A_HI);
+            switch (temp)
+            {
+                case 0x001D:
+                    return TimingBudget.Budget15;
+                case 0x0051:
+                case 0x001E:
+                    return TimingBudget.Budget20;
+                case 0x00D6:
+                case 0x0060:
+                    return TimingBudget.Budget33;
+                case 0x1AE:
+                case 0x00AD:
+                    return TimingBudget.Budget50;
+                case 0x02E1:
+                case 0x01CC:
+                    return TimingBudget.Budget100;
+                case 0x03E1:
+                case 0x02D9:
+                    return TimingBudget.Budget200;
+                case 0x0591:
+                case 0x048F:
+                    return TimingBudget.Budget500;
+                default:
+                    return TimingBudget.BudgetUnknown;
+            }
+        }
+
+        /// <summary>
+        ///     This function programs the distance mode (1 = Short, 2 = Long).
+        ///     Short mode maximum distance is limited to 1.3m but results in a better ambient immunity.
+        ///     Long mode can range up to 4 m in the dark with a timing budget of200 ms.
+        /// </summary>
+        /// <param name="mode">The mode to program</param>
+        public void SetDistanceMode(DistanceMode mode)
```

Comment:
you have a setter and a getter for the distance mode. Maybe, you should think of making it a property. And as it's not changing, it is really the definition of a property.
There are few other cases like this

Reply:
I'm now using properties wherever possible.